os/drivers/compression/compress.c

```diff
@@ -105,21 +116,86 @@ static int comp_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 		ret = CONFIG_COMPRESSION_TYPE;
 		break;
 	case COMPIOC_GET_COMP_NAME:
+		if ((char *)arg == NULL) {
+			return -EINVAL;
+		}
 		switch (CONFIG_COMPRESSION_TYPE) {
-			case LZMA_TYPE:
-				memcpy((char *)arg, LZMA_NAME, COMP_NAME_SIZE);
-				break;
-			case MINIZ_TYPE:
-				memcpy((char *)arg, MINIZ_NAME, COMP_NAME_SIZE);
-				break;
+		case LZMA_TYPE:
+			memcpy((char *)arg, LZMA_NAME, COMP_NAME_SIZE);
+			break;
+		case MINIZ_TYPE:
+			memcpy((char *)arg, MINIZ_NAME, COMP_NAME_SIZE);
+			break;
 		}
 		ret = OK;
 		break;
  	case COMPIOC_DECOMPRESS:
+		if (comp_info == NULL) {
+			return -EINVAL;
+		}
 		ret = decompress_block(comp_info->output_buffer, &comp_info->output_size, comp_info->input_buffer, &comp_info->input_size);
 		if (ret == ENOMEM) {
 			bcmpdbg("Output buffer allocated is not sufficient\n");
 		}
+	case COMPIOC_FDCMP_INIT:
+		if ((char *)arg == NULL) {
+			return -EINVAL;
+		}
+		data = (struct file_decomp_data_s *)kmm_zalloc(sizeof(struct file_decomp_data_s));
+		if (!data) {
+			bcmpdbg("Memory allocation fail for data storing\n");
+			return -ENOMEM;
+		}
+		data->fd = open((char *)arg, "r");
+		if (data->fd < 0) {
+			bcmpdbg("Failed to open file = %s\n", (char *)arg);
+			return -EINVAL;
+		}
+		DEBUGASSERT(filep);
+		filep->f_priv = data;
```

Comment:
What does the `filep` have? It's the compression path `/dev/compress`. No?
This is the point that I told not thread safe. 
If two different threads want decompress two different files, Is this working?
@kishore-sn @Sambhavi-1